packages/listbox/src/ListboxMixin.js

```diff
@@ -670,10 +673,13 @@ const ListboxMixinImplementation = superclass =>
         ev.stopPropagation();
       }
       this.__onChildCheckedChanged(ev);
-      this.requestUpdate('modelValue', this.modelValue);
+
+      // don't send this.modelValue as oldValue, since it will take modelValue getter which takes it from child elements, which is already the updated value
+      this.requestUpdate('modelValue', this.__oldModelValue);
       this.dispatchEvent(
         new CustomEvent('model-value-changed', { detail: { element: ev.target } }),
       );
+      this.__oldModelValue = this.modelValue;
```

Comment:
Does this properly trigger modelValue change on multipleChoice?

Maybe we can guard this by adding this to ListboxMixinSuite?

```js
      it('requests update for modelValue when checkedIndex changes', async () => {
        const el = await fixture(html`
        <${tag} name="gender" .modelValue=${'other'} >
          <${optionTag} .choiceValue=${'male'}></${optionTag}>
          <${optionTag} .choiceValue=${'female'}></${optionTag}>
          <${optionTag} .choiceValue=${'other'}></${optionTag}>
        </${tag}>
      `);
        expect(el.checkedIndex).to.equal(2);
        const spy = sinon.spy(el, 'requestUpdate');
        el.setCheckedIndex(1);
        expect(spy).to.have.been.calledWith('modelValue', 'other');
      });

      it('requests update for modelValue after click', async () => {
        const el = await fixture(html`
        <${tag} name="gender" .modelValue=${'other'} >
          <${optionTag} .choiceValue=${'male'}></${optionTag}>
          <${optionTag} .choiceValue=${'female'}></${optionTag}>
          <${optionTag} .choiceValue=${'other'}></${optionTag}>
        </${tag}>
      `);
        expect(el.checkedIndex).to.equal(2);
        const spy = sinon.spy(el, 'requestUpdate');
        el.formElements[0].click();
        expect(spy).to.have.been.calledWith('modelValue', 'other');
      });
```

And then also for `multipleChoice` :)

We might need this as well:

```js
        modelValue: {
          ...super.modelValue,
          type: Object,
          hasChanged: (/** @type {string|string[]} */ nV, /** @type {string|string[]} */ oV) => {
            return Array.isArray(nV) ? JSON.stringify(nV) !== JSON.stringify(oV) : nV !== oV;
          },
        },
```

Reply:
Added a commit, small fix, after that those tests pass for both single and multiple choice 👍

Reply:
Nice. I added on more remark in the test